test/utils/JsonYamlSchemaValidatorSpec.scala

```diff
@@ -0,0 +1,197 @@
+      result.map(_.spaces2) shouldBe Right(jsonStringFromFile)
+    }
+
+    "convert valid yaml api spec from file to correct json schema" in {
+
+      val yamlStringFromFile = readFile("test/resources/schemas/api_1878_schema.yaml")
+      val jsonStringFromFile = formatJson(readFile("test/resources/schemas/api_1878_schema.json"))
+
+      val result = jsonYamlSchemaValidator.yamlToJson(yamlStringFromFile)
+      result.map(_.spaces2) shouldBe Right(jsonStringFromFile)
+    }
+
+    "return a Left YamlParsingFailure when given an empty string" in {
+      val result = jsonYamlSchemaValidator.yamlToJson("")
+      result.isLeft shouldBe true
+      result shouldBe Left(YamlParsingFailure("[JsonYamlSchemaValidator][yamlToJson] Invalid YAML structure: Unexpected YAML structure"))
+    }
+
+    "return a Left YamlParsingFailure when given an invalid yaml string" in {
+      val result = jsonYamlSchemaValidator.yamlToJson("???")
+      result.isLeft shouldBe true
+      result shouldBe Left(YamlParsingFailure("[JsonYamlSchemaValidator][yamlToJson] Invalid YAML structure: Unexpected YAML structure"))
+    }
+
+    "return a Left YamlParsingFailure when given malformed yaml" in {
+      val malformedYaml =
+        """
+      name: John
+      age: thirty  # Invalid type
+      hobbies: [reading, coding
+      """
+      val result = jsonYamlSchemaValidator.yamlToJson(malformedYaml)
+      result.isLeft shouldBe true
+    }
+  }
+
+  ".jsonToYaml()" should {
+
+    "convert valid json file to yaml" in {
+
+      val yamlStringFromFile = formatYaml(readFile("test/resources/sample.yaml"))
+      val jsonStringFromFile = formatJson(readFile("test/resources/sample.json"))
+
+      val result = jsonYamlSchemaValidator.jsonToYaml(jsonStringFromFile)
+      result shouldBe Right(yamlStringFromFile)
+    }
+
+    "return a Left parsing failure when empty string" in {
+      val result: Either[ParsingFailure, String] = jsonYamlSchemaValidator.jsonToYaml("")
+      result.isLeft shouldBe true
+    }
+
+    "return a Left failure when input string is invalid" in {
+      val result: Either[ParsingFailure, String] = jsonYamlSchemaValidator.jsonToYaml("{?")
+      result.isLeft shouldBe true
+    }
+  }
+
+  ".validateJson()" should {
+
+    "return true when validating a valid json payload against the correct json schema" in {
+
+      val result = jsonYamlSchemaValidator.validateJson(api1878SchemaJson, api1878DataJson)
+
+      result shouldBe Right(ReportSuccess)
+    }
+
+    "return Left ValidationFailure when validating a invalid json payload against the desired json schema" in {
+
+      val result = jsonYamlSchemaValidator.validateJson(api1878SchemaJson, invalidJsonData)
+      result shouldBe Left(ValidationFailure("instance type (object) does not match any allowed primitive type (allowed: [\"array\"])"))
+    }
+
+    "return Left ValidationFailure when validating a empty json payload against the desired json schema" in {
```

Comment:
still a error scenario, was a mistake on my part sorry with the required section.